src/js/plugins/loader.ts

```diff
@@ -1,8 +1,15 @@
 import { PlayerError } from 'api/errors';
 import { configurePlugin, getPluginErrorCode } from 'plugins/utils';
+import type { PlayerAPI, GenericObject, PluginObj } from 'types/generic.type';
+import type SimpleModel from 'model/simplemodel';
+import type PluginModel from './model';
 
-const PluginLoader = function () {
-    this.load = function (api, pluginsModel, pluginsConfig, model) {
+export interface PluginLoaderInt {
+    load: (api: PlayerAPI, pluginsModel: PluginModel, pluginsConfig: GenericObject, model: SimpleModel) => PluginObj | PlayerError;
+}
+
+const PluginLoader = function (this: PluginLoaderInt): void {
+    this.load = function (api: PlayerAPI, pluginsModel: PluginModel, pluginsConfig: GenericObject, model: SimpleModel): PluginObj | PlayerError {
```

Comment:
Shouldn't the return type be:
`: Promise<PluginObj | PlayerError | void> {`

The `void` is for early `return Promise.resolve()`.

Reply:
From MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve

`The Promise.resolve() method returns a Promise object that is resolved with a given value`

Reply:
oh I see, yeah fixing